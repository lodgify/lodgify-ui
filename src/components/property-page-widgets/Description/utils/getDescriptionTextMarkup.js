import React, { Fragment } from 'react';
import { Button } from 'semantic-ui-react';

import { Modal } from 'elements/Modal';
import { getParagraphMarkup } from 'utils/get-paragraph-markup';

/**
 * @param  {string} descriptionText
 * @param  {string} extraDescriptionButtonText
 * @return {Object}
 */
export const getDescriptionTextMarkup = (
  descriptionText,
  extraDescriptionButtonText
) => {
  const descriptionArray = descriptionText.split(' ');

  return descriptionArray.length > 100 ? (
    <Fragment>
      {getParagraphMarkup(`${descriptionArray.slice(0, 100).join(' ')}...`)}
      <Modal
        hasPadding
        trigger={<Button basic>{extraDescriptionButtonText}</Button>}
      >
        {getParagraphMarkup(descriptionText)}
      </Modal>
    </Fragment>
  ) : (
    getParagraphMarkup(descriptionText)
  );
};
