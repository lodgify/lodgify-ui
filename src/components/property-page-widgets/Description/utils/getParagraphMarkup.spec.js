import { getDescriptionTextMarkup } from './getDescriptionTextMarkup';

describe('getDescriptionTextMarkup', () => {
  it('should return the right shape', () => {
    const descriptionText = Array(10)
      .fill('yo')
      .join(' ');
    const actual = getDescriptionTextMarkup(
      descriptionText,
      'some button text'
    );

    expect(actual).toMatchSnapshot();
  });
});
