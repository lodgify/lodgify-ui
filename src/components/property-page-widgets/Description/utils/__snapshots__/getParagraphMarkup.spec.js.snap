// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDescriptionTextMarkup should return the right shape 1`] = `
Array [
  <Paragraph
    size="medium"
    weight={null}
  >
    yo yo yo yo yo yo yo yo yo yo
  </Paragraph>,
]
`;
